internal/webhooks/webhooks.go

```diff
@@ -43,16 +45,44 @@ func Allow(msg string) admission.Response {
 	}}
 }
 
+func DenyBadRequest(err error) admission.Response {
+	return denyFromAPIError(apierrors.NewBadRequest(err.Error()))
+}
+
+func DenyConflict(qualifiedResource schema.GroupResource, name string, err error) admission.Response {
```

Comment:
I'm wondering if `name` is underspecified for most of the callers of this function (ditto for `DenyForbidden`)? E.g. an anchor name doesn't say what parent namespace it's in, and this call:

`return webhooks.DenyForbidden(api.HierarchyConfigurationGR, api.Singleton, err)`

doesn't indicate the namespace at all. Perhaps we should update the callers to either include the namespace name or accept two parameters to this function and create a string like "namespace-name/object-name" if both are provided?

Or maybe it doesn't matter because the namespace names are already in the error messages? wdyt?

Reply:
I understand your concern here, and for sure the error messages and internal API for creating the errors can be improved. We could use [NamespacedName](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apimachinery/pkg/types/namespacedname.go#L27) for namespaced resources, but for cluster resources that does not make sense. Currently these new API error factory methods are just delegating into the corresponding function in api-machinery.

I personally try to avoid having multiple parameters of the same type in functions, mainly because I think positional parameters are error-prone. But that is a personal preference. 😅 

Since this PR is a refactor PR, I would prefer leaving it as is for now. I never had the ambition to make all error messages "perfect" here, but I am quite sure I am not making the error messages worse in this PR!

Reply:
Fair enough, thanks.